Tests/SKSwiftPMWorkspaceTests/SwiftPMBuildSystemTests.swift

```diff
@@ -179,6 +183,52 @@ final class SwiftPMBuildSystemTests: XCTestCase {
     }
   }
 
+  func testCompilerArgumentsForFileThatContainsPlusCharacterURLEncoded() async throws {
+    try await withTestScratchDir { tempDir in
+      try localFileSystem.createFiles(
+        root: tempDir,
+        files: [
+          "pkg/Sources/lib/a.swift": "",
+          "pkg/Sources/lib/a+something.swift": "",
+          "pkg/Package.swift": """
+          // swift-tools-version:4.2
+          import PackageDescription
+          let package = Package(
+            name: "a",
+            targets: [.target(name: "lib")]
+          )
+          """,
+        ]
+      )
+      let packageRoot = try resolveSymlinks(tempDir.appending(component: "pkg"))
+      let tr = ToolchainRegistry.forTesting
+      let swiftpmBuildSystem = try await SwiftPMBuildSystem(
+        workspacePath: packageRoot,
+        toolchainRegistry: tr,
+        fileSystem: localFileSystem,
+        buildSetup: SourceKitLSPServer.Options.testDefault.buildSetup,
+        isForIndexBuild: false
+      )
+      try await swiftpmBuildSystem.generateBuildGraph(allowFileSystemWrites: false)
+
+      let aPlusSomething = packageRoot.appending(components: "Sources", "lib", "a+something.swift")
+      print(aPlusSomething.asURL)
+      let hostTriple = await swiftpmBuildSystem.destinationBuildParameters.triple
+      let build = buildPath(root: packageRoot, platform: hostTriple.platformBuildPathComponent)
+
+      assertEqual(await swiftpmBuildSystem.buildPath, build)
+      assertNotNil(await swiftpmBuildSystem.indexStorePath)
+      let arguments = try await unwrap(
+        swiftpmBuildSystem.buildSettings(
+          for: DocumentURI(URL(string: "file://\(aPlusSomething.asURL.path.replacing("+", with: "%2B"))")!),
+          language: .swift
+        )
+      )
+      .compilerArguments
+      print(arguments)
```

Comment:
Left over from debugging?

Reply:
Yes, I guess I should add an actual check 🙈